src/osd/ReplicatedPG.cc

```diff
@@ -3505,6 +3505,23 @@ static string list_entries(const T& m) {
   return s;
 }
 
+bool ReplicatedPG::maybe_create_new_object(OpContext *ctx)
+{
+  ObjectState& obs = ctx->new_obs;
+  if (!obs.exists) {
+    ctx->mod_desc.create();
```

Comment:
The ops below often manipulate mod_desk before calling maybe_create_new_object.  We probably shouldn't do this here.